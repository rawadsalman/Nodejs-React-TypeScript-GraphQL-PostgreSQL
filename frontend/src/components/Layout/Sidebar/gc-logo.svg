<svg width="21" height="21" viewBox="0 0 107 128" version="1.1" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;">
    <g id="Page-1">
        <g id="_6">
            <g id="iconmonstr-antenna-5">
                <g>
                    <path id="Shape" d="M0,128L15.772,128L34.764,117.333L53.757,128L69.529,128L34.764,95.999L0,128ZM107,85.107C107,102.543 75.66,103.172 54.64,94.654C18.072,79.832 3.124,42.439 12.819,13.744C17.313,0.431 22.727,-6.737 46.7,8.826C52.591,12.645 58.845,17.622 65.088,23.344L55.886,29.489C14.6,-7.894 18.479,18.63 50.627,53.757C81.967,87.982 113.051,99.705 81.481,57.581L87.895,48.274C94.438,56.989 107,74.264 107,85.107ZM101.511,18.774C101.49,23.195 97.873,26.758 93.443,26.731C92.239,26.726 91.111,26.433 90.083,25.953L69.598,56.066C65.404,53.106 58.93,46.636 55.966,42.444L86.365,22.336C85.809,21.232 85.461,20.006 85.466,18.688C85.493,14.267 89.104,10.704 93.529,10.73C97.959,10.757 101.538,14.357 101.511,18.774Z" style="fill-rule:nonzero;"/>
                </g>
            </g>
        </g>
    </g>
</svg>
