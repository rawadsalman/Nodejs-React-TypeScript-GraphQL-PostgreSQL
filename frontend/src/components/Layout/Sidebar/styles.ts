import { keyframes } from '@emotion/react'
import styled from '@emotion/styled'

const FlexBoxDiv = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  height: 100%;
  box-shadow: 1px 0px 10px 1px rgba(0, 0, 0, 0.77);
`

const shake = keyframes`
  0% { transform: translate(1px, 1px) rotate(0deg); }
  10% { transform: translate(-1px, -2px) rotate(-1deg); }
  20% { transform: translate(-3px, 0px) rotate(1deg); }
  30% { transform: translate(3px, 2px) rotate(0deg); }
  40% { transform: translate(1px, -1px) rotate(1deg); }
  50% { transform: translate(-1px, 2px) rotate(-1deg); }
  60% { transform: translate(-3px, 1px) rotate(0deg); }
  70% { transform: translate(3px, 1px) rotate(-1deg); }
  80% { transform: translate(-1px, -1px) rotate(1deg); }
  90% { transform: translate(1px, 2px) rotate(0deg); }
  100% { transform: translate(1px, -2px) rotate(-1deg); }
`

const TitleDiv = styled.div`
  min-height: 4rem;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  hr {
    margin-top: 0;
    width: 70%;
  }
`

const LogoName = styled.h2`
  font-family: Raleway;
  color: white;
  font-size: 1.1rem;
  line-height: 0.8em;
  display: block;

  &:hover {
    animation: ${shake} 500ms ease infinite;
  }
`

export { FlexBoxDiv, TitleDiv, LogoName }
