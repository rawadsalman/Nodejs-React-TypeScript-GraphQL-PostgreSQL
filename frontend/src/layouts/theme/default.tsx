// rgb(17, 24, 57) blau
// rgb(247, 230, 153)	gelb

const defaultTheme = {
  colors: {
    primary: 'rgb(255, 0, 130)',
    secondary: '#1e272e',
    green: '#16a085',
    darkBlue: '#001529',
    almostWhite: '#f0f0f0',
    almostGrey: '#eaeaea',
    yellow: '#f6e58d',
    paleRed: '#ff7675',
  },
  modal: {
    width: 720,
  },
}

export default defaultTheme
