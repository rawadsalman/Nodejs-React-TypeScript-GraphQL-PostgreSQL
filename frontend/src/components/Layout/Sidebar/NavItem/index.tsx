import React from 'react'
import { useHistory, useRouteMatch } from 'react-router-dom'
import { NavItemFlyout, NavItemInnerDiv, NavItemOuterDiv } from './styles'

interface INavItem {
  to: string
  activeOnlyWhenExact?: boolean
  displayName?: string
}

const NavItem: React.FC<INavItem> = ({ children, to, activeOnlyWhenExact, displayName }) => {
  let match = useRouteMatch({
    path: to,
    exact: activeOnlyWhenExact,
  })
  let history = useHistory()

  const handleClick = () => history.push(to)

  return (
    <NavItemOuterDiv className={`${match ? 'active' : ''}`} onClick={handleClick}>
      <NavItemInnerDiv className={`${match ? 'active' : ''}`}>{children}</NavItemInnerDiv>
      <NavItemFlyout className="flyout">
        <span>{displayName ? displayName : to.substr(1)}</span>
      </NavItemFlyout>
    </NavItemOuterDiv>
  )
}

export default NavItem
