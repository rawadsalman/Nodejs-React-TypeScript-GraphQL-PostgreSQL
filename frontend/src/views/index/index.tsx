import React from 'react'
import { StyledContent } from '../styles'

const IndexView: React.FC = () => {
  return (
    <StyledContent>
      <p>Hey there</p>
    </StyledContent>
  )
}

export default IndexView
