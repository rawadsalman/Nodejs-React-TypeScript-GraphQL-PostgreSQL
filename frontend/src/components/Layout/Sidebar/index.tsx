import React from 'react'
import { Link } from 'react-router-dom'
import { FlexBoxDiv, LogoName, TitleDiv } from './styles'
import NavItem from './NavItem'
import { ReactComponent as CompanyIcon } from './../../Icon/iconmonstr-building-46.svg'

const Sidebar: React.FC = () => {
  return (
    <FlexBoxDiv>
      <div>
        <TitleDiv>
          <Link to="/">
            <LogoName>
              Space
              <br />
              Goats
            </LogoName>
          </Link>
          <hr />
        </TitleDiv>

        <NavItem to="/companies">
          <CompanyIcon />
        </NavItem>
      </div>
    </FlexBoxDiv>
  )
}

export default Sidebar
