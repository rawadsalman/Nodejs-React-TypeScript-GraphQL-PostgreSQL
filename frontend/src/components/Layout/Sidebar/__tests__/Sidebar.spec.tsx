import { render } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Sidebar from '../index'
import { ThemeProvider } from '@emotion/react'
import { defaultTheme } from '../../../../layouts/theme/index'

describe('Sidebar', () => {
  it('matches Snapshot', async () => {
    const wrapper = render(
      <ThemeProvider theme={defaultTheme}>
        <Router>
          <Sidebar />
        </Router>
      </ThemeProvider>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
