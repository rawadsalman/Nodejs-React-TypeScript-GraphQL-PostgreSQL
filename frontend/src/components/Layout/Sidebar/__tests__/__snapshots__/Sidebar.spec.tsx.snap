// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-17qfunh-FlexBoxDiv e1x7gwan2"
      >
        <div>
          <div
            class="css-va9r8i-TitleDiv e1x7gwan1"
          >
            <a
              href="/"
            >
              <h2
                class="css-1br8yq8-LogoName e1x7gwan0"
              >
                Space
                <br />
                Goats
              </h2>
            </a>
            <hr />
          </div>
          <div
            class=" css-1imfco6-NavItemOuterDiv ek60gb91"
          >
            <div
              class=" css-m82ati-NavItemInnerDiv ek60gb92"
            >
              <svg>
                iconmonstr-building-46.svg
              </svg>
            </div>
            <div
              class="flyout css-1ev6f10-NavItemFlyout ek60gb90"
            >
              <span>
                companies
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-17qfunh-FlexBoxDiv e1x7gwan2"
    >
      <div>
        <div
          class="css-va9r8i-TitleDiv e1x7gwan1"
        >
          <a
            href="/"
          >
            <h2
              class="css-1br8yq8-LogoName e1x7gwan0"
            >
              Space
              <br />
              Goats
            </h2>
          </a>
          <hr />
        </div>
        <div
          class=" css-1imfco6-NavItemOuterDiv ek60gb91"
        >
          <div
            class=" css-m82ati-NavItemInnerDiv ek60gb92"
          >
            <svg>
              iconmonstr-building-46.svg
            </svg>
          </div>
          <div
            class="flyout css-1ev6f10-NavItemFlyout ek60gb90"
          >
            <span>
              companies
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
